sdk/storage/Azure.Storage.Blobs/src/Generated/BlobRestClient.cs

```diff
@@ -17843,53 +17906,127 @@ public static partial class BlobsModelFactory
         /// The SHA-256 hash of the encryption key used to encrypt the pages. This header is only returned when the pages were encrypted with a customer-provided key.
         /// </summary>
-        public string EncryptionKeySha256 { get; internal set; }
+        public string EncryptionKeySha256 { get; }
 
         /// <summary>
         /// Prevent direct instantiation of PageInfo instances.
         /// You can use BlobsModelFactory.PageInfo instead.
         /// </summary>
-        internal PageInfo() { }
+        internal PageInfo(
+            Azure.Core.Http.ETag eTag,
+            System.DateTimeOffset lastModified,
+            byte[] contentHash,
+            byte[] xMSContentCrc64,
+            long blobSequenceNumber,
+            string encryptionKeySha256){
+                ETag = eTag;
+                LastModified = lastModified;
+                ContentHash = contentHash;
+                XMSContentCrc64 = xMSContentCrc64;
+                BlobSequenceNumber = blobSequenceNumber;
+                EncryptionKeySha256 = encryptionKeySha256;
+            }
+
+        /// <summary>
+        /// Check if two PageInfo instances are equal.
+        /// </summary>
+        /// <param name="other">The instance to compare to.</param>
+        /// <returns>True if they're equal, false otherwise.</returns>
+        public bool Equals(PageInfo other)
+        {
+            if (!ETag.Equals(other.ETag))
+            {
+                return false;
+            }
+            if (!LastModified.Equals(other.LastModified))
+            {
+                return false;
+            }
+            if (!Equals(ContentHash, other.ContentHash))
+            {
+                return false;
+            }
+            if (!Equals(XMSContentCrc64, other.XMSContentCrc64))
+            {
+                return false;
+            }
+            if (!BlobSequenceNumber.Equals(other.BlobSequenceNumber))
+            {
+                return false;
+            }
+            if (!System.StringComparer.Ordinal.Equals(EncryptionKeySha256, other.EncryptionKeySha256))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Check if two PageInfo instances are equal.
+        /// </summary>
+        /// <param name="obj">The instance to compare to.</param>
+        /// <returns>True if they're equal, false otherwise.</returns>
+        public override bool Equals(object obj) => obj is PageInfo && Equals((PageInfo)obj);
+
+        /// <summary>
+        /// Get a hash code for the PageInfo.
+        /// </summary>
+        public override int GetHashCode()
+        {
+            var hashCode = new Azure.Core.HashCodeBuilder();
+            hashCode.Add(ETag);
+            hashCode.Add(LastModified);
+            hashCode.Add(ContentHash);
+            hashCode.Add(XMSContentCrc64);
+            hashCode.Add(BlobSequenceNumber);
+            if (EncryptionKeySha256 != null)
```

Comment:
`XMSContentCrc64` and `ContentHash` can be `null` as well.

Reply:
If `XMSContentCrc64` or `ContentHash` is null then [Add(value)](https://github.com/Azure/azure-sdk-for-net/blob/d3d19ff0057bb3443cf39929b8e79d3ba208bd85/sdk/core/Azure.Core/src/Shared/HashCodeBuilder.cs#L262) method will set the value 0.

Reply:
I've also written a test case to verify the [null case scenarios](https://github.com/Azure/azure-sdk-for-net/blob/a99fe024bbef0e5771013a358920063821f8bc6b/sdk/storage/Azure.Storage.Blobs/tests/PageInfoTest.cs#L42).

Reply:
I've a test case to verify the [null case scenarios](https://github.com/Azure/azure-sdk-for-net/blob/a99fe024bbef0e5771013a358920063821f8bc6b/sdk/storage/Azure.Storage.Blobs/tests/PageInfoTest.cs#L42).